### ISSUE TITLE
Inserts can be lost during a predicate move

### ISSUE THREAD

**Issue by aphyr** [dgraph-io/dgraph#2338]
With server-side ordering, @upsert schemas, no crashes or network faults, roughly 10 inserts/sec, and no updates or deletes, Dgraph can occasionally (once every five hours or so) lose successfully inserted records: [20180412T161038.000-0500.zip](https://github.com/dgraph-io/dgraph/files/1916026/20180412T161038.000-0500.zip)

```
 {:valid? false,
  :lost "#{10185}",
  :recovered "#{}",
  :ok
  "#{0..1 5..6 8 10 12..13 15 18 20 22 24 ... 74997 75000..75001}"
  :recovered-frac 0,
  :unexpected-frac 0,
  :unexpected "#{}",
  :lost-frac 1/75002,
  :ok-frac 16645/37501}
```

The lost records occur during a predicate move, which suggests this issue might be related to #2321. This occurs with

```
Dgraph version   : v1.0.4
Commit SHA-1     : 6fb69e2
Commit timestamp : 2018-04-09 21:26:31 +0530
Branch           : jan/node_lockup
```

and can be reproduced with Jepsen 23329ead4c4e3d8352234658026d09792f15c406 via

```
lein run test --package-url https://transfer.sh/BAPac/dgraph-linux-amd64.tar.gz --time-limit 3600 --concurrency 1n --nemesis none --test-count 20 --workload set --upsert-schema
```

**Comment by aphyr** [comment]
You can now reproduce this problem significantly faster in Jepsen eb796cfcc204c592545965968bd28ad1e6b2eff0 by using the move-tablet nemesis, which shuffles tablets around every 15 seconds or so.

```
lein run test --package-url https://github.com/dgraph-io/dgraph/releases/download/nightly/dgraph-linux-amd64.tar.gz --time-limit 60 --concurrency 1n --nemesis move-tablet --workload set --upsert-schema
```

**Comment by janardhan1993** [comment]
@aphyr You were probably using old nightly build,  did you try with `--force-download` flag.

**Comment by aphyr** [comment]
Ah, good catch, thank you. Unfortunately we're still broken in b8c09088. :-(

**Comment by janardhan1993** [comment]
Some of the issue are fixed in #2339

**Comment by manishrjain** [comment]
Based on binary off mrjn/fix-2321.

```lein run test --time-limit 1800  --concurrency 1n --nemesis move-tablet --workload set --upsert-schema --local-binary /gobin/dgraph --force-download --retry-db-setup --test-count 3```

![2018-05-18-121847_896x81_scrot](https://user-images.githubusercontent.com/1782324/40253726-b0b49e2c-5a95-11e8-839c-8408462e595a.png)

Fuck yeah!
